evalml/pipelines/utils.py

```diff
@@ -844,9 +848,34 @@ def _make_new_component_name(name, component_name, idx=None, pipeline_name=None)
         final_components.append(final_component)
 
     final_y = final_y_candidate if final_y_candidate else final_y
-    component_graph[estimator.name] = (
-        [estimator] + [comp + ".x" for comp in final_components] + [final_y]
-    )
+    last_pre_estimator_component = None
+    if post_pipelines_components:
+        first_pre_estimator_component = (
+            list(post_pipelines_components.keys())[0]
+            if post_pipelines_components
+            else None
```

Comment:
Isn't this first post_pipelines_components.keys()[0] guaranteed to exist?  You're already inside a `if post_pipelines_components:` branch here.